website/docs/configuration/types.html.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: "docs"
+page_title: "Type Constraints - Configuration Language"
+sidebar_current: "docs-config-types"
+description: |-
+  Terraform module authors and provider developers can use detailed type
+  constraints to validate the inputs of their modules and resources.
+---
+
+# Type Constraints
```

Comment:
(Could use special attention to this whole page, but a few specific things called out below.)